make/premake5.lua

```diff
@@ -0,0 +1,52 @@
+--  ============================================================
+--  premake5 build rules
+--  ============================================================
+
+buildOptsGcc = '-std=c++17 -pedantic-errors -Werror -Wall '
+buildOptsVS = '/std:c++17 /W4 ' 
+
+--  ============================================================
+--  sample TCP_Srv_Echo based on TCP_Srv_Base
+--  with tracing off
+--  ============================================================
+workspace 'TCP_Srv_Echo_silent'
+    configurations { 'ci' }
+    language 'C++'
+    objdir 'obj/%{prj.name}'
+
+    project 'TCP_Srv_Echo_silent'
+        kind 'ConsoleApp'
+        files { '../code/*.cpp' }
+
+        filter { 'action:vs*' }
+            warnings 'high'
+            buildoptions { buildOptsVS .. '/wd4100' }
+            targetdir 'exe'
+    
+        filter { 'action:gmake*' }
+            buildoptions { buildOptsGcc }
+            linkoptions { '-pthread' }
+            targetdir 'bin'
+
+--  ============================================================
+--  sample TCP_Srv_Echo based on TCP_Srv_Base
+--  with tracing on
+--  ============================================================
+workspace 'TCP_Srv_Echo_verbose'
```

Comment:
remove second version
- name the whole thing TCP_Srv_Echo

Reply:
done

Reply:
now it's called _EchoSrv_